docs/DeveloperGuide.md

```diff
@@ -1,38 +1,312 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#ui-component)
+    * [Parser Component](#parser-component)
+    * [Recipe Component](#recipe-component)
+    * [RecipeList Component](#recipelist-component)
+    * [WeeklyPlan Component](#weeklyplan-component)
+    * [Database Component](#database-component)
+* [Implementation](#implementation)
+    * [Add Recipes Feature](#add-recipes-feature)
+    * [Edit Recipes Feature](#edit-recipes-feature)
+    * [Categorise/Tag Recipes Feature](#categorisetag-recipes-feature)
+    * [List Recipes Feature](#list-recipes-feature)
+    * [Delete Recipes Feature](#delete-recipes-feature)
+
+* [Appendix: Requirements](#appendix-requirements)
+    * [Product scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
+---
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries --
+include links to the original source as well}
+
+---
+
+## Setting up, getting started
+
+---
+
+## Design
+
+* [Architecture](#architecture)
+* [UI Component](#ui-component)
+* [Parser Component](#parser-component)
+* [Recipe Component](#recipe-component)
+* [RecipeList Component](#recipelist-component)
+* [WeeklyPlan Component](#weeklyplan-component)
+* [Database Component](#database-component)
```

Comment:
I like that you link the subpoints under each subheader, however it may be redundant as you have already added it in the table of content at the start and adding this again would be confusing. Perhaps you could considering either removing this part or the subparts in the table of content at the start.